replication/backup.go

```diff
@@ -13,7 +13,26 @@ import (
 
 // StartBackup: Like mysqlbinlog remote raw backup
 // Backup remote binlog from position (filename, offset) and write in backupDir
-func (b *BinlogSyncer) StartBackup(backupDir string, p Position, timeout time.Duration) error {
+func (b *BinlogSyncer) StartBackupToFile(backupDir string, p Position, timeout time.Duration) error {
+	err := os.MkdirAll(backupDir, 0755)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	return b.StartBackup(p, timeout, func(filename string) (io.WriteCloser, error) {
+		return os.OpenFile(path.Join(backupDir, filename), os.O_CREATE|os.O_WRONLY, 0644)
+	})
+}
+
+// StartBackup starts the backup process for the binary log using the specified position and handler.
+// The process will continue until the timeout is reached or an error occurs.
+//
+// Parameters:
+//   - p: The starting position in the binlog from which to begin the backup.
+//   - timeout: The maximum duration to wait for new binlog events before stopping the backup process.
+//     If set to 0, a default very long timeout (30 days) is used instead.
+//   - handler: A function that takes a binlog filename and returns an WriteCloser for writing raw events to.
+func (b *BinlogSyncer) StartBackup(p Position, timeout time.Duration,
```

Comment:
Would you consider to rename this function to `StartBackupWithHandler`, and let the existing function keep the name `StartBackup`? Just for the compatibility. 

It's not a problem because other PRs also modify the public APIs and we can make a major version release to declare the breaking changes, but I think renaming your new function is also acceptable? Please leave your opinions.

rest lgtm

Reply:
Renaming this function is acceptable for compatibility purposes